pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -217,12 +219,27 @@ func filterSeedsMatchingProviders(cloudProfile *gardencorev1beta1.CloudProfile,
 	return matchingSeeds, nil
 }
 
+// filterHighAvailabilitySeeds filters seeds based on their high-availability configurations
+func filterHighAvailabilitySeeds(seedList []gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) ([]gardencorev1beta1.Seed, error) {
+	if gardencorev1beta1helper.IsHAControlPlaneConfigured(shoot) {
+		var highAvailabilitySeeds []gardencorev1beta1.Seed
+		for _, seed := range seedList {
+			if gardencorev1beta1helper.IsHASeedConfigured(&seed) {
+				highAvailabilitySeeds = append(highAvailabilitySeeds, seed)
+			}
+		}
+		if len(highAvailabilitySeeds) == 0 {
+			return nil, fmt.Errorf("none of the %d seeds can host a high-availability control plane for the given shoot", len(seedList))
+		}
+		return highAvailabilitySeeds, nil
+	}
+	// for any shoot that is non-HA
+	return seedList, nil
+}
+
 // filterSeedsMatchingMultiZonalAttribute filters seeds by label `seed.gardener.cloud/multi-zonal`,
 // based on whether the shoot requires a multi-zonal control plane or not
 func filterSeedsMatchingMultiZonalAttribute(seedList []gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) ([]gardencorev1beta1.Seed, error) {
-	if !gardenerschedulerfeatures.FeatureGate.Enabled(features.HAControlPlanes) {
-		return seedList, nil
-	}
```

Comment:
As far as I see this is the only usage of the `HAControlPlanes` feature gate in the gardener-scheduler. Can you elaborate why we remove this check now?
If we preserve this, I think we have to clean up also https://github.com/gardener/gardener/blob/18bb09016bb0dd61d5fabdb7a3012220c34d4b5d/docs/deployment/feature_gates.md#L159
and usages of this feature gate in the unit tests.

Reply:
This filter removes seeds which are not multi zonal from the set of seeds when the shoot has failure tolerance "zone".
The feature flag can disable this filter, so that seeds which are not multi zonal remain in the set of potential seeds for this shoot.
I do not see any scenario where this is useful, because [admission-controller prevents it anyway].(https://github.com/gardener/gardener/blob/8f48339c5b5be7ca96072dbe680b9e7935ebe811/plugin/pkg/shoot/validator/admission.go#L360-L365).

Reply:
As discussed, fine for me to remove the feature gate from the scheduler. Please also add a proper release note.